// ngx-bar-rating/esm2020/lib/bar-rating.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { BarRating } from './bar-rating';
import { BarRatingPipe } from './bar-rating.pipe';
import { ActiveRating, InactiveRating, FractionRating } from './custom-rating';
import * as i0 from "@angular/core";
export class BarRatingModule {
}
BarRatingModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: BarRatingModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
BarRatingModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: BarRatingModule, declarations: [BarRating,
        BarRatingPipe,
        ActiveRating,
        InactiveRating,
        FractionRating], imports: [CommonModule,
        FormsModule], exports: [BarRating,
        ActiveRating,
        InactiveRating,
        FractionRating] });
BarRatingModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: BarRatingModule, imports: [[
            CommonModule,
            FormsModule
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.1.3", ngImport: i0, type: BarRatingModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        BarRating,
                        BarRatingPipe,
                        ActiveRating,
                        InactiveRating,
                        FractionRating
                    ],
                    imports: [
                        CommonModule,
                        FormsModule
                    ],
                    exports: [
                        BarRating,
                        ActiveRating,
                        InactiveRating,
                        FractionRating
                    ]
                }]
        }] });
