// ng-lazyload-image/esm2020/src/lazyload-image.module.mjs
import { NgModule } from '@angular/core';
import { LazyLoadImageDirective } from './lazyload-image.directive';
import { IntersectionObserverHooks } from './intersection-observer-hooks/hooks';
import { LAZYLOAD_IMAGE_HOOKS } from './token';
import * as i0 from "@angular/core";
export class LazyLoadImageModule {
}
LazyLoadImageModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.1.1", ngImport: i0, type: LazyLoadImageModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
LazyLoadImageModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.1.1", ngImport: i0, type: LazyLoadImageModule, declarations: [LazyLoadImageDirective], exports: [LazyLoadImageDirective] });
LazyLoadImageModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.1.1", ngImport: i0, type: LazyLoadImageModule, providers: [{ provide: LAZYLOAD_IMAGE_HOOKS, useClass: IntersectionObserverHooks }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.1.1", ngImport: i0, type: LazyLoadImageModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LazyLoadImageDirective],
                    exports: [LazyLoadImageDirective],
                    providers: [{ provide: LAZYLOAD_IMAGE_HOOKS, useClass: IntersectionObserverHooks }],
                }]
        }] });
