// ng-animate/bundles/ng-animate.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/animations')) :
    typeof define === 'function' && define.amd ? define('ng-animate', ['exports', '@angular/animations'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory(global['ng-animate'] = {}, global.ng.animations));
}(this, (function (exports, animations) { 'use strict';

    var DEFAULT_TIMING = 1;

    var bounce = animations.animation([
        animations.style({ transform: 'translate3d(0, 0, 0)' }),
        animations.animate('{{ timing }}s {{ delay }}s', animations.keyframes([
            animations.style({ transform: 'translate3d(0, 0, 0)', offset: 0.2 }),
            animations.style({ transform: 'translate3d(0, -30px, 0)', offset: 0.4 }),
            animations.style({ transform: 'translate3d(0, 0, 0)', offset: 0.53 }),
            animations.style({ transform: 'translate3d(0, -15px, 0)', offset: 0.7 }),
            animations.style({ transform: 'translate3d(0, -4px, 0)', offset: 0.9 }),
            animations.style({ transform: 'translate3d(0, 0, 0)', offset: 1 }),
        ])),
    ], { params: { timing: DEFAULT_TIMING, delay: 0 } });
    var flash = animations.animation(animations.animate('{{ timing }}s {{ delay }}s', animations.keyframes([
        animations.style({ opacity: 1 }),
        animations.style({ opacity: 0 }),
        animations.style({ opacity: 1 }),
        animations.style({ opacity: 0 }),
        animations.style({ opacity: 1 }),
    ])), { params: { timing: DEFAULT_TIMING, delay: 0 } });
    var pulse = animations.animation(animations.animate('{{ timing }}s {{ delay }}s', animations.keyframes([
        animations.style({ transform: 'scale3d(1, 1, 1)' }),
        animations.style({ transform: 'scale3d({{ scale }}, {{ scale }}, {{ scale }})' }),
        animations.style({ transform: 'scale3d(1, 1, 1)' }),
    ])), { params: { scale: 1.25, timing: DEFAULT_TIMING, delay: 0 } });
    var rubberBand = animations.animation(animations.animate('{{ timing }}s {{ delay }}s', animations.keyframes([
        animations.style({ transform: 'scale3d(1, 1, 1)', offset: 0 }),
        animations.style({ transform: 'scale3d(1.25, 0.75, 1)', offset: 0.3 }),
        animations.style({ transform: 'scale3d(0.75, 1.25, 1)', offset: 0.4 }),
        animations.style({ transform: 'scale3d(1.15, 0.85, 1)', offset: 0.5 }),
        animations.style({ transform: 'scale3d(.95, 1.05, 1)', offset: 0.65 }),
        animations.style({ transform: 'scale3d(1.05, .95, 1)', offset: 0.75 }),
        animations.style({ transform: 'scale3d(1, 1, 1)', offset: 1 }),
    ])), { params: { timing: DEFAULT_TIMING, delay: 0 } });
    var shake = animations.animation(animations.animate('{{ timing }}s {{ delay }}s', animations.keyframes([
        animations.style({ transform: 'translate3d(0, 0, 0)', offset: 0 }),
        animations.style({ transform: 'translate3d(-10px, 0, 0)', offset: 0.1 }),
        animations.style({ transform: 'translate3d(10px, 0, 0)', offset: 0.2 }),
        animations.style({ transform: 'translate3d(-10px, 0, 0)', offset: 0.3 }),
        animations.style({ transform: 'translate3d(10px, 0, 0)', offset: 0.4 }),
        animations.style({ transform: 'translate3d(-10px, 0, 0)', offset: 0.5 }),
        animations.style({ transform: 'translate3d(10px, 0, 0)', offset: 0.6 }),
        animations.style({ transform: 'translate3d(-10px, 0, 0)', offset: 0.7 }),
        animations.style({ transform: 'translate3d(10px, 0, 0)', offset: 0.8 }),
        animations.style({ transform: 'translate3d(-10px, 0, 0)', offset: 0.9 }),
        animations.style({ transform: 'translate3d(0, 0, 0)', offset: 1 }),
    ])), { params: { timing: DEFAULT_TIMING, delay: 0 } });
    var swing = animations.animation(animations.animate('{{ timing }}s {{ delay }}s', animations.keyframes([
        animations.style({ transform: 'rotate3d(0, 0, 1, 15deg)', offset: 0.2 }),
        animations.style({ transform: 'rotate3d(0, 0, 1, -10deg)', offset: 0.4 }),
        animations.style({ transform: 'rotate3d(0, 0, 1, 5deg)', offset: 0.6 }),
        animations.style({ transform: 'rotate3d(0, 0, 1, -5deg)', offset: 0.8 }),
        animations.style({ transform: 'rotate3d(0, 0, 1, 0deg)', offset: 1 }),
    ])), { params: { timing: DEFAULT_TIMING, delay: 0 } });
    var tada = animations.animation(animations.animate('{{ timing }}s {{ delay }}s', animations.keyframes([
        animations.style({ transform: 'scale3d(1, 1, 1)', offset: 0 }),
        animations.style({
            transform: 'scale3d(.9, .9, .9) rotate3d(0, 0, 1, -3deg)',
            offset: 0.1,
        }),
        animations.style({
            transform: 'scale3d(.9, .9, .9) rotate3d(0, 0, 1, -3deg)',
            offset: 0.2,
        }),
        animations.style({
            transform: 'scale3d(1.1, 1.1, 1.1) rotate3d(0, 0, 1, 3deg)',
            offset: 0.3,
        }),
        animations.style({
            transform: 'scale3d(1.1, 1.1, 1.1) rotate3d(0, 0, 1, -3deg)',
            offset: 0.4,
        }),
        animations.style({
            transform: 'scale3d(1.1, 1.1, 1.1) rotate3d(0, 0, 1, 3deg)',
            offset: 0.5,
        }),
        animations.style({
            transform: 'scale3d(1.1, 1.1, 1.1) rotate3d(0, 0, 1, -3deg)',
            offset: 0.6,
        }),
        animations.style({
            transform: 'scale3d(1.1, 1.1, 1.1) rotate3d(0, 0, 1, 3deg)',
            offset: 0.7,
        }),
        animations.style({
            transform: 'scale3d(1.1, 1.1, 1.1) rotate3d(0, 0, 1, -3deg)',
            offset: 0.8,
        }),
        animations.style({
            transform: 'scale3d(1.1, 1.1, 1.1) rotate3d(0, 0, 1, 3deg)',
            offset: 0.9,
        }),
        animations.style({ transform: 'scale3d(1, 1, 1)', offset: 1 }),
    ])), { params: { timing: DEFAULT_TIMING, delay: 0 } });
    var wobble = animations.animation(animations.animate('{{ timing }}s {{ delay }}s', animations.keyframes([
        animations.style({ transform: 'none', offset: 0 }),
        animations.style({
            transform: 'translate3d(-25%, 0, 0) rotate3d(0, 0, 1, -5deg)',
            offset: 0.15,
        }),
        animations.style({
            transform: 'translate3d(20%, 0, 0) rotate3d(0, 0, 1, 3deg)',
            offset: 0.3,
        }),
        animations.style({
            transform: 'translate3d(-15%, 0, 0) rotate3d(0, 0, 1, -3deg)',
            offset: 0.45,
        }),
        animations.style({
            transform: 'translate3d(10%, 0, 0) rotate3d(0, 0, 1, 2deg)',
            offset: 0.6,
        }),
        animations.style({
            transform: 'translate3d(-5%, 0, 0) rotate3d(0, 0, 1, -1deg)',
            offset: 0.75,
        }),
        animations.style({ transform: 'none', offset: 1 }),
    ])), { params: { timing: DEFAULT_TIMING, delay: 0 } });
    var jello = animations.animation(animations.animate('{{ timing }}s {{ delay }}s', animations.keyframes([
        animations.style({ transform: 'none', offset: 0 }),
        animations.style({ transform: 'none', offset: 0.11 }),
        animations.style({ transform: 'skewX(-12.5deg) skewY(-12.5deg)', offset: 0.22 }),
        animations.style({ transform: 'skewX(6.25deg) skewY(6.25deg)', offset: 0.33 }),
        animations.style({ transform: 'skewX(-3.125deg) skewY(-3.125deg)', offset: 0.44 }),
        animations.style({ transform: 'skewX(1.5625deg) skewY(1.5625deg)', offset: 0.55 }),
        animations.style({
            transform: 'skewX(-0.78125deg) skewY(-0.78125deg)',
            offset: 0.66,
        }),
        animations.style({
            transform: 'skewX(0.390625deg) skewY(0.390625deg)',
            offset: 0.77,
        }),
        animations.style({
            transform: 'skewX(-0.1953125deg) skewY(-0.1953125deg)',
            offset: 0.88,
        }),
        animations.style({ transform: 'none', offset: 1 }),
    ])), { params: { timing: DEFAULT_TIMING, delay: 0 } });

    var bounceIn = animations.animation(animations.animate('{{ timing }}s {{ delay }}s cubic-bezier(0.215, 0.610, 0.355, 1.000)', animations.keyframes([
        animations.style({ opacity: 0, transform: 'scale3d(.3, .3, .3)', offset: 0 }),
        animations.style({ transform: 'scale3d(1.1, 1.1, 1.1)', offset: 0.2 }),
        animations.style({ transform: 'scale3d(.9, .9, .9)', offset: 0.4 }),
        animations.style({
            opacity: 1,
            transform: 'scale3d(1.03, 1.03, 1.03)',
            offset: 0.6
        }),
        animations.style({ transform: 'scale3d(.97, .97, .97)', offset: 0.8 }),
        animations.style({ opacity: 1, transform: 'scale3d(1, 1, 1)', offset: 1 })
    ])), { params: { timing: DEFAULT_TIMING, delay: 0 } });
    function bounceInY(a, b, c, d) {
        return animations.animation(animations.animate('{{ timing }}s {{ delay }}s cubic-bezier(0.215, 0.610, 0.355, 1.000)', animations.keyframes([
            animations.style({
                opacity: 0,
                transform: 'translate3d(0, {{ a }}, 0)',
                offset: 0
            }),
            animations.style({
                opacity: 1,
                transform: 'translate3d(0, {{ b }}, 0)',
                offset: 0.6
            }),
            animations.style({ transform: 'translate3d(0, {{ c }}, 0)', offset: 0.75 }),
            animations.style({ transform: 'translate3d(0, {{ d }}, 0)', offset: 0.9 }),
            animations.style({ opacity: 1, transform: 'none', offset: 1 })
        ])), {
            params: {
                timing: DEFAULT_TIMING,
                delay: 0,
                a: a,
                b: b,
                c: c,
                d: d
            }
        });
    }
    function bounceInX(a, b, c, d) {
        return animations.animation(animations.animate('{{ timing }}s {{ delay }}s cubic-bezier(0.215, 0.610, 0.355, 1.000)', animations.keyframes([
            animations.style({
                opacity: 0,
                transform: 'translate3d({{ a }}, 0, 0)',
                offset: 0
            }),
            animations.style({
                opacity: 1,
                transform: 'translate3d({{ b }}, 0, 0)',
                offset: 0.6
            }),
            animations.style({ transform: 'translate3d({{ c }}, 0, 0)', offset: 0.75 }),
            animations.style({ transform: 'translate3d({{ d }}, 0, 0)', offset: 0.9 }),
            animations.style({ opacity: 1, transform: 'none', offset: 1 })
        ])), {
            params: {
                timing: DEFAULT_TIMING,
                delay: 0,
                a: a,
                b: b,
                c: c,
                d: d
            }
        });
    }
    var bounceInDown = bounceInY('-3000px', '25px', '-10px', '5px');
    var bounceInUp = bounceInY('3000px', '-25px', '10px', '-5px');
    var bounceInLeft = bounceInX('-3000px', '25px', '-10px', '5px');
    var bounceInRight = bounceInX('3000px', '-25px', '10px', '-5px');
    var bounceOut = animations.animation(animations.animate('{{ timing }}s {{ delay }}s', animations.keyframes([
        animations.style({ transform: 'scale3d(.9, .9, .9)', offset: 0.2 }),
        animations.style({
            opacity: 1,
            transform: 'scale3d({{ scale }}, {{ scale }}, {{ scale }})',
            offset: 0.5
        }),
        animations.style({
            opacity: 1,
            transform: 'scale3d({{ scale }}, {{ scale }}, {{ scale }})',
            offset: 0.55
        }),
        animations.style({ opacity: 0, transform: 'scale3d(.3, .3, .3)', offset: 1 })
    ])), { params: { timing: DEFAULT_TIMING, delay: 0, scale: 1.1 } });
    function bounceOutY(a, b, c, d) {
        return animations.animation(animations.animate('{{ timing }}s {{ delay }}s', animations.keyframes([
            animations.style({ transform: 'translate3d(0, {{ a }}, 0)', offset: 0.2 }),
            animations.style({
                opacity: 1,
                transform: 'translate3d(0, {{ b }}, 0)',
                offset: 0.4
            }),
            animations.style({
                opacity: 1,
                transform: 'translate3d(0, {{ c }}, 0)',
                offset: 0.45
            }),
            animations.style({
                opacity: 0,
                transform: 'translate3d(0, {{ d }}, 0)',
                offset: 1
            })
        ])), {
            params: {
                timing: DEFAULT_TIMING,
                delay: 0,
                a: a,
                b: b,
                c: c,
                d: d
            }
        });
    }
    function bounceOutX(a, b) {
        return animations.animation(animations.animate('{{ timing }}s {{ delay }}s', animations.keyframes([
            animations.style({
                opacity: 1,
                transform: 'translate3d({{ a }}, 0, 0)',
                offset: 0.2
            }),
            animations.style({
                opacity: 0,
                transform: 'translate3d({{ b }}, 0, 0)',
                offset: 1
            })
        ])), { params: { timing: DEFAULT_TIMING, delay: 0, a: a, b: b } });
    }
    var bounceOutDown = bounceOutY('10px', '-20px', '-20px', '2000px');
    var bounceOutUp = bounceOutY('-10px', '20px', '20px', '-2000px');
    var bounceOutLeft = bounceOutX('20px', '-2000px');
    var bounceOutRight = bounceOutX('-20px', '2000px');

    function fadeInX(a, b) {
        return animations.animation(animations.animate('{{ timing }}s {{ delay }}s', animations.keyframes([
            animations.style({
                opacity: 0,
                transform: 'translate3d({{ a }}, 0, 0)',
                offset: 0
            }),
            animations.style({
                opacity: 1,
                transform: 'translate3d({{ b }}, 0, 0)',
                offset: 1
            })
        ])), { params: { timing: DEFAULT_TIMING, delay: 0, a: a, b: b } });
    }
    function fadeInY(a, b) {
        return animations.animation(animations.animate('{{ timing }}s {{ delay }}s', animations.keyframes([
            animations.style({
                opacity: 0,
                transform: 'translate3d(0, {{ a }}, 0)',
                offset: 0
            }),
            animations.style({
                opacity: 1,
                transform: 'translate3d(0, {{ b }}, 0)',
                offset: 1
            })
        ])), { params: { timing: DEFAULT_TIMING, delay: 0, a: a, b: b } });
    }
    var fadeIn = fadeInX(0, 0);
    var fadeInDown = fadeInY('-100%', 0);
    var fadeInUp = fadeInY('100%', 0);
    var fadeInLeft = fadeInX('-100%', 0);
    var fadeInRight = fadeInX('100%', 0);
    function fadeOutX(a, b) {
        return animations.animation(animations.animate('{{ timing }}s {{ delay }}s', animations.keyframes([
            animations.style({
                opacity: 1,
                transform: 'translate3d({{ a }}, 0, 0)',
                offset: 0
            }),
            animations.style({
                opacity: 0,
                transform: 'translate3d({{ b }}, 0, 0)',
                offset: 1
            })
        ])), { params: { timing: DEFAULT_TIMING, delay: 0, a: a, b: b } });
    }
    function fadeOutY(a, b) {
        return animations.animation(animations.animate('{{ timing }}s {{ delay }}s', animations.keyframes([
            animations.style({
                opacity: 1,
                transform: 'translate3d(0, {{ a }}, 0)',
                offset: 0
            }),
            animations.style({
                opacity: 0,
                transform: 'translate3d(0, {{ b }}, 0)',
                offset: 1
            })
        ])), { params: { timing: DEFAULT_TIMING, delay: 0, a: a, b: b } });
    }
    var fadeOut = fadeOutX(0, 0);
    var fadeOutDown = fadeOutY('-100%', 0);
    var fadeOutUp = fadeOutY('100%', 0);
    var fadeOutLeft = fadeOutX('-100%', 0);
    var fadeOutRight = fadeOutX('100%', 0);
    function slideX(a, b) {
        return animations.animation(animations.animate('{{ timing }}s {{ delay }}s', animations.keyframes([
            animations.style({
                transform: 'translate3d({{ a }}, 0, 0)',
                offset: 0
            }),
            animations.style({
                transform: 'translate3d({{ b }}, 0, 0)',
                offset: 1
            })
        ])), { params: { timing: DEFAULT_TIMING, delay: 0, a: a, b: b } });
    }
    function slideY(a, b) {
        return animations.animation(animations.animate('{{ timing }}s {{ delay }}s', animations.keyframes([
            animations.style({
                transform: 'translate3d(0, {{ a }}, 0)',
                offset: 0
            }),
            animations.style({
                transform: 'translate3d(0, {{ b }}, 0)',
                offset: 1
            })
        ])), { params: { timing: DEFAULT_TIMING, delay: 0, a: a, b: b } });
    }
    var slideInUp = slideY('-100%', 0);
    var slideInDown = slideY('100%', 0);
    var slideInLeft = slideX('-100%', 0);
    var slideInRight = slideX('100%', 0);
    var slideOutUp = slideY(0, '-100%');
    var slideOutDown = slideY(0, '100%');
    var slideOutLeft = slideX(0, '-100%');
    var slideOutRight = slideX(0, '100%');

    var flip = animations.animation([
        animations.style({ 'backface-visibility': 'visible' }),
        animations.animate('{{ timing }}s {{ delay }}s ease-out', animations.keyframes([
            animations.style({
                transform: 'perspective(400px) rotate3d(0, 1, 0, -360deg)',
                offset: 0
            }),
            animations.style({
                transform: 'perspective(400px) scale3d(1.5, 1.5, 1.5) rotate3d(0, 1, 0, -190deg)',
                offset: 0.4
            }),
            animations.style({
                transform: 'perspective(400px) scale3d(1.5, 1.5, 1.5) rotate3d(0, 1, 0, -170deg)',
                offset: 0.5
            }),
            animations.style({
                transform: 'perspective(400px) scale3d(.95, .95, .95)',
                offset: 0.8
            }),
            animations.style({
                transform: 'perspective(400px)',
                offset: 1
            })
        ]))
    ], {
        params: { timing: DEFAULT_TIMING, delay: 0 }
    });
    function flipIn(rotateX, rotateY) {
        return animations.animation([
            animations.style({ 'backface-visibility': 'visible' }),
            animations.animate('{{ timing }}s {{ delay }}s ease-in', animations.keyframes([
                animations.style({
                    opacity: 0,
                    transform: 'perspective(400px) rotate3d({{ rotateX }}, {{ rotateY }}, 0, 90deg)',
                    offset: 0
                }),
                animations.style({
                    opacity: 1,
                    transform: 'perspective(400px) rotate3d({{ rotateX }}, {{ rotateY }}, 0, -20deg)',
                    offset: 0.4
                }),
                animations.style({
                    transform: 'perspective(400px) rotate3d({{ rotateX }}, {{ rotateY }}, 0, 10deg)',
                    offset: 0.6
                }),
                animations.style({
                    transform: 'perspective(400px) rotate3d({{ rotateX }}, {{ rotateY }}, 0, -5deg)',
                    offset: 0.8
                }),
                animations.style({
                    transform: 'perspective(400px) rotate3d(0, 0, 0, 0)',
                    offset: 1
                })
            ]))
        ], { params: { timing: DEFAULT_TIMING, delay: 0, rotateX: rotateX, rotateY: rotateY } });
    }
    var flipInX = flipIn(1, 0);
    var flipInY = flipIn(0, 1);
    function flipOut(rotateX, rotateY) {
        return animations.animation([
            animations.style({ 'backface-visibility': 'visible' }),
            animations.animate('{{ timing }}s {{ delay }}s', animations.keyframes([
                animations.style({
                    transform: 'perspective(400px)',
                    offset: 0
                }),
                animations.style({
                    opacity: 1,
                    transform: 'perspective(400px) rotate3d({{ rotateX }}, {{ rotateY }}, 0, -20deg)',
                    offset: 0.3
                }),
                animations.style({
                    opacity: 0,
                    transform: 'perspective(400px) rotate3d({{ rotateX }}, {{ rotateY }}, 0, 90deg)',
                    offset: 1
                })
            ]))
        ], { params: { timing: DEFAULT_TIMING, delay: 0, rotateX: rotateX, rotateY: rotateY } });
    }
    var flipOutX = flipOut(1, 0);
    var flipOutY = flipOut(0, 1);

    var lightSpeedIn = animations.animation(animations.animate('{{ timing }}s {{ delay }}s', animations.keyframes([
        animations.style({
            opacity: 0,
            transform: 'translate3d(100%, 0, 0) skewX(-30deg)',
            offset: 0,
        }),
        animations.style({
            opacity: 1,
            transform: 'translate3d(0, 0, 0) skewX(0)',
            offset: 1,
        }),
    ])), {
        params: { timing: DEFAULT_TIMING, delay: 0 },
    });
    var lightSpeedOut = animations.animation(animations.animate('{{ timing }}s {{ delay }}s ease-out', animations.keyframes([
        animations.style({
            opacity: 1,
            offset: 0,
        }),
        animations.style({
            opacity: 0,
            transform: 'translate3d(100%, 0, 0) skewX(30deg)',
            offset: 1,
        }),
    ])), {
        params: { timing: DEFAULT_TIMING, delay: 0 },
    });

    function rotateInDirection(origin, degrees) {
        return animations.animation(animations.animate('{{ timing }}s {{ delay }}s', animations.keyframes([
            animations.style({
                'transform-origin': '{{ origin }}',
                opacity: 0,
                transform: 'rotate3d(0, 0, 1, {{ degrees }})',
                offset: 0
            }),
            animations.style({
                'transform-origin': '{{ origin }}',
                opacity: 1,
                transform: 'none',
                offset: 1
            })
        ])), {
            params: { timing: DEFAULT_TIMING, delay: 0, origin: origin, degrees: degrees }
        });
    }
    var rotateIn = rotateInDirection('center', '-200deg');
    var rotateInDownLeft = rotateInDirection('left bottom', '-45deg');
    var rotateInDownRight = rotateInDirection('right bottom', '45deg');
    var rotateInUpLeft = rotateInDirection('left bottom', '45deg');
    var rotateInUpRight = rotateInDirection('right bottom', '-90deg');
    function rotateOutDirection(origin, degrees) {
        return animations.animation(animations.animate('{{ timing }}s {{ delay }}s', animations.keyframes([
            animations.style({
                'transform-origin': '{{ origin }}',
                opacity: 1,
                transform: 'none',
                offset: 0
            }),
            animations.style({
                'transform-origin': '{{ origin }}',
                opacity: 0,
                transform: 'rotate3d(0, 0, 1, {{ degrees }})',
                offset: 1
            })
        ])), {
            params: { timing: DEFAULT_TIMING, delay: 0, origin: origin, degrees: degrees }
        });
    }
    var rotateOut = rotateOutDirection('center', '200deg');
    var rotateOutDownLeft = rotateOutDirection('left bottom', '45deg');
    var rotateOutDownRight = rotateOutDirection('right bottom', '-45deg');
    var rotateOutUpLeft = rotateOutDirection('left bottom', '-45deg');
    var rotateOutUpRight = rotateOutDirection('right bottom', '90deg');

    var hinge = animations.animation([
        animations.style({ 'transform-origin': 'top left' }),
        animations.animate('{{ timing }}s {{ delay }}s ease-in-out', animations.keyframes([
            animations.style({
                transform: 'rotate3d(0, 0, 1, 80deg)',
                offset: 0.2,
            }),
            animations.style({
                transform: 'rotate3d(0, 0, 1, 60deg)',
                offset: 0.4,
            }),
            animations.style({
                transform: 'rotate3d(0, 0, 1, 80deg)',
                offset: 0.6,
            }),
            animations.style({
                opacity: 1,
                transform: 'rotate3d(0, 0, 1, 60deg)',
                offset: 0.8,
            }),
            animations.style({
                opacity: 0,
                transform: 'translate3d(0, 700px, 0)',
                offset: 1,
            }),
        ])),
    ], {
        params: { timing: DEFAULT_TIMING, delay: 0 },
    });
    var jackInTheBox = animations.animation([
        animations.animate('{{ timing }}s {{ delay }}s', animations.keyframes([
            animations.style({
                opacity: 0,
                transform: 'scale(0.1) rotate(30deg)',
                'transform-origin': 'center bottom',
                offset: 0,
            }),
            animations.style({
                opacity: 0.5,
                transform: 'rotate(-10deg)',
                offset: 0.5,
            }),
            animations.style({
                opacity: 0.7,
                transform: 'rotate(3deg)',
                offset: 0.7,
            }),
            animations.style({
                opacity: 1,
                transform: 'scale(1)',
                offset: 1,
            }),
        ])),
    ], {
        params: { timing: DEFAULT_TIMING, delay: 0 },
    });
    var rollIn = animations.animation([
        animations.animate('{{ timing }}s {{ delay }}s', animations.keyframes([
            animations.style({
                opacity: 0,
                transform: 'translate3d(-100%, 0, 0) rotate3d(0, 0, 1, -120deg)',
                offset: 0,
            }),
            animations.style({
                opacity: 1,
                transform: 'none',
                offset: 1,
            }),
        ])),
    ], {
        params: { timing: DEFAULT_TIMING, delay: 0 },
    });
    var rollOut = animations.animation([
        animations.animate('{{ timing }}s {{ delay }}s', animations.keyframes([
            animations.style({
                opacity: 1,
                offset: 0,
            }),
            animations.style({
                opacity: 0,
                transform: 'translate3d(100%, 0, 0) rotate3d(0, 0, 1, 120deg)',
                offset: 1,
            }),
        ])),
    ], {
        params: { timing: DEFAULT_TIMING, delay: 0 },
    });

    var zoomIn = animations.animation([
        animations.animate('{{ timing }}s {{ delay }}s', animations.keyframes([
            animations.style({
                opacity: 0,
                transform: 'scale3d(.3, .3, .3)',
                offset: 0
            }),
            animations.style({
                opacity: 1,
                transform: 'scale3d(1, 1, 1)',
                offset: 0.5
            })
        ]))
    ], {
        params: { timing: DEFAULT_TIMING, delay: 0 }
    });
    function zoomInY(a, b) {
        return animations.animation(animations.animate('{{ timing }}s {{ delay }}s cubic-bezier(0.550, 0.055, 0.675, 0.190)', animations.keyframes([
            animations.style({
                opacity: 0,
                transform: "scale3d(.1, .1, .1) translate3d(0, {{ a }}, 0)",
                offset: 0
            }),
            animations.style({
                opacity: 1,
                transform: "scale3d(.475, .475, .475) translate3d(0, {{ b }}, 0)",
                offset: 0.6
            })
        ])), { params: { timing: DEFAULT_TIMING, delay: 0, a: a, b: b } });
    }
    function zoomInX(a, b) {
        return animations.animation(animations.animate('{{ timing }}s {{ delay }}s cubic-bezier(0.550, 0.055, 0.675, 0.190)', animations.keyframes([
            animations.style({
                opacity: 0,
                transform: "scale3d(.1, .1, .1) translate3d({{ a }}, 0, 0)",
                offset: 0
            }),
            animations.style({
                opacity: 1,
                transform: "scale3d(.475, .475, .475) translate3d({{ b }}, 0, 0)",
                offset: 0.6
            })
        ])), { params: { timing: DEFAULT_TIMING, delay: 0, a: a, b: b } });
    }
    var zoomInDown = zoomInY('-1000px', '10px');
    var zoomInUp = zoomInY('1000px', '-10px');
    var zoomInLeft = zoomInX('-1000px', '10px');
    var zoomInRight = zoomInX('1000px', '-10px');
    var zoomOut = animations.animation([
        animations.animate('{{ timing }}s {{ delay }}s', animations.keyframes([
            animations.style({
                opacity: 1,
                offset: 0
            }),
            animations.style({
                opacity: 0,
                transform: 'scale3d(.3, .3, .3)',
                offset: 0.5
            }),
            animations.style({
                opacity: 0,
                offset: 1
            })
        ]))
    ], {
        params: { timing: DEFAULT_TIMING, delay: 0 }
    });
    function zoomOutY(a, b) {
        return animations.animation(animations.animate('{{ timing }}s {{ delay }}s cubic-bezier(0.550, 0.055, 0.675, 0.190)', animations.keyframes([
            animations.style({
                opacity: 1,
                transform: "scale3d(.475, .475, .475) translate3d(0, {{ a }}, 0)",
                offset: 0.4
            }),
            animations.style({
                opacity: 0,
                transform: "scale3d(.1, .1, .1) translate3d(0, {{ b }}, 0)",
                offset: 1
            })
        ])), { params: { timing: DEFAULT_TIMING, delay: 0, a: a, b: b } });
    }
    function zoomOutX(a, b) {
        return animations.animation(animations.animate('{{ timing }}s {{ delay }}s cubic-bezier(0.550, 0.055, 0.675, 0.190)', animations.keyframes([
            animations.style({
                opacity: 1,
                transform: "scale3d(.475, .475, .475) translate3d({{ a }}, 0, 0)",
                offset: 0.4
            }),
            animations.style({
                opacity: 0,
                transform: "scale3d(.1, .1, .1) translate3d({{ b }}, 0, 0)",
                offset: 1
            })
        ])), { params: { timing: DEFAULT_TIMING, delay: 0, a: a, b: b } });
    }
    var zoomOutDown = zoomOutY('-60px', '2000px');
    var zoomOutUp = zoomOutY('60px', '-2000px');
    var zoomOutLeft = zoomOutX('42px', '-2000px');
    var zoomOutRight = zoomOutX('-42px', '2000px');

    /*
     * Public API Surface of ng-animate
     */

    /**
     * Generated bundle index. Do not edit.
     */

    exports.bounce = bounce;
    exports.bounceIn = bounceIn;
    exports.bounceInDown = bounceInDown;
    exports.bounceInLeft = bounceInLeft;
    exports.bounceInRight = bounceInRight;
    exports.bounceInUp = bounceInUp;
    exports.bounceInX = bounceInX;
    exports.bounceInY = bounceInY;
    exports.bounceOut = bounceOut;
    exports.bounceOutDown = bounceOutDown;
    exports.bounceOutLeft = bounceOutLeft;
    exports.bounceOutRight = bounceOutRight;
    exports.bounceOutUp = bounceOutUp;
    exports.bounceOutX = bounceOutX;
    exports.bounceOutY = bounceOutY;
    exports.fadeIn = fadeIn;
    exports.fadeInDown = fadeInDown;
    exports.fadeInLeft = fadeInLeft;
    exports.fadeInRight = fadeInRight;
    exports.fadeInUp = fadeInUp;
    exports.fadeInX = fadeInX;
    exports.fadeInY = fadeInY;
    exports.fadeOut = fadeOut;
    exports.fadeOutDown = fadeOutDown;
    exports.fadeOutLeft = fadeOutLeft;
    exports.fadeOutRight = fadeOutRight;
    exports.fadeOutUp = fadeOutUp;
    exports.fadeOutX = fadeOutX;
    exports.fadeOutY = fadeOutY;
    exports.flash = flash;
    exports.flip = flip;
    exports.flipIn = flipIn;
    exports.flipInX = flipInX;
    exports.flipInY = flipInY;
    exports.flipOut = flipOut;
    exports.flipOutX = flipOutX;
    exports.flipOutY = flipOutY;
    exports.hinge = hinge;
    exports.jackInTheBox = jackInTheBox;
    exports.jello = jello;
    exports.lightSpeedIn = lightSpeedIn;
    exports.lightSpeedOut = lightSpeedOut;
    exports.pulse = pulse;
    exports.rollIn = rollIn;
    exports.rollOut = rollOut;
    exports.rotateIn = rotateIn;
    exports.rotateInDirection = rotateInDirection;
    exports.rotateInDownLeft = rotateInDownLeft;
    exports.rotateInDownRight = rotateInDownRight;
    exports.rotateInUpLeft = rotateInUpLeft;
    exports.rotateInUpRight = rotateInUpRight;
    exports.rotateOut = rotateOut;
    exports.rotateOutDirection = rotateOutDirection;
    exports.rotateOutDownLeft = rotateOutDownLeft;
    exports.rotateOutDownRight = rotateOutDownRight;
    exports.rotateOutUpLeft = rotateOutUpLeft;
    exports.rotateOutUpRight = rotateOutUpRight;
    exports.rubberBand = rubberBand;
    exports.shake = shake;
    exports.slideInDown = slideInDown;
    exports.slideInLeft = slideInLeft;
    exports.slideInRight = slideInRight;
    exports.slideInUp = slideInUp;
    exports.slideOutDown = slideOutDown;
    exports.slideOutLeft = slideOutLeft;
    exports.slideOutRight = slideOutRight;
    exports.slideOutUp = slideOutUp;
    exports.slideX = slideX;
    exports.slideY = slideY;
    exports.swing = swing;
    exports.tada = tada;
    exports.wobble = wobble;
    exports.zoomIn = zoomIn;
    exports.zoomInDown = zoomInDown;
    exports.zoomInLeft = zoomInLeft;
    exports.zoomInRight = zoomInRight;
    exports.zoomInUp = zoomInUp;
    exports.zoomInX = zoomInX;
    exports.zoomInY = zoomInY;
    exports.zoomOut = zoomOut;
    exports.zoomOutDown = zoomOutDown;
    exports.zoomOutLeft = zoomOutLeft;
    exports.zoomOutRight = zoomOutRight;
    exports.zoomOutUp = zoomOutUp;
    exports.zoomOutX = zoomOutX;
    exports.zoomOutY = zoomOutY;
    exports.ɵa = DEFAULT_TIMING;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
