// ngx-masonry/bundles/ngx-masonry.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/common'), require('@angular/core'), require('@angular/animations')) :
    typeof define === 'function' && define.amd ? define('ngx-masonry', ['exports', '@angular/common', '@angular/core', '@angular/animations'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory(global['ngx-masonry'] = {}, global.ng.common, global.ng.core, global.ng.animations));
}(this, (function (exports, common, core, animations) { 'use strict';

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation.

    Permission to use, copy, modify, and/or distribute this software for any
    purpose with or without fee is hereby granted.

    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
    REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
    AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
    INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
    LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
    OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
    PERFORMANCE OF THIS SOFTWARE.
    ***************************************************************************** */
    /* global Reflect, Promise */
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b)
                if (Object.prototype.hasOwnProperty.call(b, p))
                    d[p] = b[p]; };
        return extendStatics(d, b);
    };
    function __extends(d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }
    var __assign = function () {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s)
                    if (Object.prototype.hasOwnProperty.call(s, p))
                        t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };
    function __rest(s, e) {
        var t = {};
        for (var p in s)
            if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
                t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }
    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
            r = Reflect.decorate(decorators, target, key, desc);
        else
            for (var i = decorators.length - 1; i >= 0; i--)
                if (d = decorators[i])
                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }
    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); };
    }
    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
            return Reflect.metadata(metadataKey, metadataValue);
    }
    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try {
                step(generator.next(value));
            }
            catch (e) {
                reject(e);
            } }
            function rejected(value) { try {
                step(generator["throw"](value));
            }
            catch (e) {
                reject(e);
            } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }
    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function () { if (t[0] & 1)
                throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function () { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f)
                throw new TypeError("Generator is already executing.");
            while (_)
                try {
                    if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done)
                        return t;
                    if (y = 0, t)
                        op = [op[0] & 2, t.value];
                    switch (op[0]) {
                        case 0:
                        case 1:
                            t = op;
                            break;
                        case 4:
                            _.label++;
                            return { value: op[1], done: false };
                        case 5:
                            _.label++;
                            y = op[1];
                            op = [0];
                            continue;
                        case 7:
                            op = _.ops.pop();
                            _.trys.pop();
                            continue;
                        default:
                            if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {
                                _ = 0;
                                continue;
                            }
                            if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) {
                                _.label = op[1];
                                break;
                            }
                            if (op[0] === 6 && _.label < t[1]) {
                                _.label = t[1];
                                t = op;
                                break;
                            }
                            if (t && _.label < t[2]) {
                                _.label = t[2];
                                _.ops.push(op);
                                break;
                            }
                            if (t[2])
                                _.ops.pop();
                            _.trys.pop();
                            continue;
                    }
                    op = body.call(thisArg, _);
                }
                catch (e) {
                    op = [6, e];
                    y = 0;
                }
                finally {
                    f = t = 0;
                }
            if (op[0] & 5)
                throw op[1];
            return { value: op[0] ? op[1] : void 0, done: true };
        }
    }
    var __createBinding = Object.create ? (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        Object.defineProperty(o, k2, { enumerable: true, get: function () { return m[k]; } });
    }) : (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        o[k2] = m[k];
    });
    function __exportStar(m, o) {
        for (var p in m)
            if (p !== "default" && !Object.prototype.hasOwnProperty.call(o, p))
                __createBinding(o, m, p);
    }
    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m)
            return m.call(o);
        if (o && typeof o.length === "number")
            return {
                next: function () {
                    if (o && i >= o.length)
                        o = void 0;
                    return { value: o && o[i++], done: !o };
                }
            };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }
    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m)
            return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done)
                ar.push(r.value);
        }
        catch (error) {
            e = { error: error };
        }
        finally {
            try {
                if (r && !r.done && (m = i["return"]))
                    m.call(i);
            }
            finally {
                if (e)
                    throw e.error;
            }
        }
        return ar;
    }
    /** @deprecated */
    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }
    /** @deprecated */
    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++)
            s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    }
    function __spreadArray(to, from, pack) {
        if (pack || arguments.length === 2)
            for (var i = 0, l = from.length, ar; i < l; i++) {
                if (ar || !(i in from)) {
                    if (!ar)
                        ar = Array.prototype.slice.call(from, 0, i);
                    ar[i] = from[i];
                }
            }
        return to.concat(ar || Array.prototype.slice.call(from));
    }
    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }
    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n])
            i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try {
            step(g[n](v));
        }
        catch (e) {
            settle(q[0][3], e);
        } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length)
            resume(q[0][0], q[0][1]); }
    }
    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }
    function __asyncValues(o) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function (v) { resolve({ value: v, done: d }); }, reject); }
    }
    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) {
            Object.defineProperty(cooked, "raw", { value: raw });
        }
        else {
            cooked.raw = raw;
        }
        return cooked;
    }
    ;
    var __setModuleDefault = Object.create ? (function (o, v) {
        Object.defineProperty(o, "default", { enumerable: true, value: v });
    }) : function (o, v) {
        o["default"] = v;
    };
    function __importStar(mod) {
        if (mod && mod.__esModule)
            return mod;
        var result = {};
        if (mod != null)
            for (var k in mod)
                if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k))
                    __createBinding(result, mod, k);
        __setModuleDefault(result, mod);
        return result;
    }
    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }
    function __classPrivateFieldGet(receiver, state, kind, f) {
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a getter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot read private member from an object whose class did not declare it");
        return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
    }
    function __classPrivateFieldSet(receiver, state, value, kind, f) {
        if (kind === "m")
            throw new TypeError("Private method is not writable");
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a setter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot write private member to an object whose class did not declare it");
        return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
    }

    var masonryConstructor;
    var NgxMasonryComponent = /** @class */ (function () {
        function NgxMasonryComponent(platformId, _element) {
            this.platformId = platformId;
            this._element = _element;
            this.updateLayout = false;
            this.ordered = false;
            // Outputs
            this.layoutComplete = new core.EventEmitter();
            this.removeComplete = new core.EventEmitter();
            this.itemsLoaded = new core.EventEmitter();
            this.pendingItems = [];
        }
        NgxMasonryComponent.prototype.ngOnInit = function () {
            var _this = this;
            if (common.isPlatformBrowser(this.platformId) && masonryConstructor === undefined) {
                masonryConstructor = require('masonry-layout');
            }
            // Create masonry options object
            if (!this.options) {
                this.options = {};
            }
            // Set default itemSelector
            if (!this.options.itemSelector) {
                this.options.itemSelector = '[ngxMasonryItem], ngxMasonryItem';
            }
            this.options['transitionDuration'] = '0s';
            if (common.isPlatformBrowser(this.platformId)) {
                // Initialize Masonry
                this.masonryInstance = new masonryConstructor(this._element.nativeElement, this.options);
                // Bind to events
                this.masonryInstance.on('layoutComplete', function (items) {
                    _this.layoutComplete.emit(items);
                });
                this.masonryInstance.on('removeComplete', function (items) {
                    _this.removeComplete.emit(items);
                });
                this.masonryInstance.items = [];
            }
        };
        NgxMasonryComponent.prototype.ngOnChanges = function (changes) {
            // only update layout if it's not the first change
            if (changes.updateLayout) {
                if (!changes.updateLayout.firstChange) {
                    this.layout();
                }
            }
        };
        NgxMasonryComponent.prototype.ngOnDestroy = function () {
            if (this.masonryInstance) {
                this.masonryInstance.destroy();
            }
        };
        NgxMasonryComponent.prototype.layout = function () {
            var _this = this;
            setTimeout(function () {
                _this.masonryInstance.layout();
            });
        };
        NgxMasonryComponent.prototype.reloadItems = function () {
            var _this = this;
            setTimeout(function () {
                _this.masonryInstance.reloadItems();
            });
        };
        NgxMasonryComponent.prototype.addPendingItem = function (item) {
            this.pendingItems.push(item);
        };
        NgxMasonryComponent.prototype.add = function (newItem) {
            var e_1, _a;
            if (this.ordered) {
                try {
                    for (var _b = __values(this.pendingItems.entries()), _c = _b.next(); !_c.done; _c = _b.next()) {
                        var _d = __read(_c.value, 2), index = _d[0], item = _d[1];
                        if (item) {
                            if (item.images && item.images.size === 0) {
                                this.pendingItems[index] = undefined;
                                this.itemLoaded(item);
                                if (index + 1 === this.pendingItems.length) {
                                    // All items are loaded
                                    this.itemsLoaded.emit(this.pendingItems.length);
                                    this.pendingItems = [];
                                }
                            }
                            else {
                                return;
                            }
                        }
                    }
                }
                catch (e_1_1) { e_1 = { error: e_1_1 }; }
                finally {
                    try {
                        if (_c && !_c.done && (_a = _b.return)) _a.call(_b);
                    }
                    finally { if (e_1) throw e_1.error; }
                }
            }
            else {
                this.itemLoaded(newItem);
            }
        };
        NgxMasonryComponent.prototype.itemLoaded = function (item) {
            if (common.isPlatformBrowser(this.platformId)) {
                // Tell Masonry that a child element has been added
                if (item.prepend) {
                    this.masonryInstance.prepended(item.element.nativeElement);
                }
                else {
                    this.masonryInstance.appended(item.element.nativeElement);
                }
                // Check if first item
                if (this.masonryInstance.items.length === 1) {
                    this.masonryInstance.layout();
                }
                item.playAnimation(true);
            }
        };
        NgxMasonryComponent.prototype.remove = function (element) {
            if (common.isPlatformBrowser(this.platformId)) {
                // Tell Masonry that a child element has been removed
                this.masonryInstance.remove(element);
                // Layout items
                this.layout();
            }
        };
        return NgxMasonryComponent;
    }());
    NgxMasonryComponent.decorators = [
        { type: core.Component, args: [{
                    selector: '[ngx-masonry], ngx-masonry',
                    template: '<ng-content></ng-content>',
                    styles: ["\n\t\t:host {\n\t\t\tdisplay: block;\n\t\t}\n\t"]
                },] }
    ];
    NgxMasonryComponent.ctorParameters = function () { return [
        { type: undefined, decorators: [{ type: core.Inject, args: [core.PLATFORM_ID,] }] },
        { type: core.ElementRef }
    ]; };
    NgxMasonryComponent.propDecorators = {
        options: [{ type: core.Input }],
        updateLayout: [{ type: core.Input }],
        ordered: [{ type: core.Input }],
        layoutComplete: [{ type: core.Output }],
        removeComplete: [{ type: core.Output }],
        itemsLoaded: [{ type: core.Output }]
    };

    var NgxMasonryDirective = /** @class */ (function () {
        function NgxMasonryDirective(element, builder, parent, renderer) {
            this.element = element;
            this.builder = builder;
            this.parent = parent;
            this.renderer = renderer;
            this.prepend = false;
            this.animations = {
                show: [
                    animations.style({ opacity: 0 }),
                    animations.animate('400ms ease-in', animations.style({ opacity: 1 })),
                ],
                hide: [
                    animations.style({ opacity: '*' }),
                    animations.animate('400ms ease-in', animations.style({ opacity: 0 })),
                ]
            };
        }
        NgxMasonryDirective.prototype.ngOnInit = function () {
            if (this.parent.options.animations !== undefined) {
                this.animations = this.parent.options.animations;
            }
            this.renderer.setStyle(this.element.nativeElement, 'position', 'fixed');
            this.renderer.setStyle(this.element.nativeElement, 'right', '-150vw');
            this.parent.addPendingItem(this);
        };
        NgxMasonryDirective.prototype.ngAfterViewInit = function () {
            var e_1, _a;
            var _this = this;
            var images = Array.from(this.element.nativeElement.getElementsByTagName('img'));
            this.images = new Set(images);
            if (images.length === 0) {
                setTimeout(function () {
                    _this.parent.add(_this);
                });
            }
            else {
                var _loop_1 = function (imageRef) {
                    // skip image render check if image has `masonryLazy` attribute
                    if (imageRef.hasAttribute('masonryLazy')) {
                        this_1.imageLoaded(imageRef);
                    }
                    else {
                        this_1.renderer.listen(imageRef, 'load', function (_) {
                            _this.imageLoaded(imageRef);
                        });
                        this_1.renderer.listen(imageRef, 'error', function (_) {
                            _this.imageLoaded(imageRef);
                        });
                    }
                };
                var this_1 = this;
                try {
                    for (var images_1 = __values(images), images_1_1 = images_1.next(); !images_1_1.done; images_1_1 = images_1.next()) {
                        var imageRef = images_1_1.value;
                        _loop_1(imageRef);
                    }
                }
                catch (e_1_1) { e_1 = { error: e_1_1 }; }
                finally {
                    try {
                        if (images_1_1 && !images_1_1.done && (_a = images_1.return)) _a.call(images_1);
                    }
                    finally { if (e_1) throw e_1.error; }
                }
            }
        };
        NgxMasonryDirective.prototype.ngOnDestroy = function () {
            if (this.images && this.images.size === 0 && this.element.nativeElement.parentNode) {
                this.playAnimation(false);
                this.parent.remove(this.element.nativeElement);
            }
        };
        NgxMasonryDirective.prototype.imageLoaded = function (image) {
            this.images.delete(image);
            if (this.images.size === 0) {
                this.parent.add(this);
            }
        };
        NgxMasonryDirective.prototype.playAnimation = function (show) {
            var metadata = show ? this.animations.show : this.animations.hide;
            if (metadata) {
                var player = this.builder.build(metadata).create(this.element.nativeElement);
                player.play();
            }
        };
        return NgxMasonryDirective;
    }());
    NgxMasonryDirective.decorators = [
        { type: core.Directive, args: [{
                    selector: '[ngxMasonryItem], ngxMasonryItem'
                },] }
    ];
    NgxMasonryDirective.ctorParameters = function () { return [
        { type: core.ElementRef },
        { type: animations.AnimationBuilder },
        { type: NgxMasonryComponent, decorators: [{ type: core.Inject, args: [core.forwardRef(function () { return NgxMasonryComponent; }),] }] },
        { type: core.Renderer2 }
    ]; };
    NgxMasonryDirective.propDecorators = {
        prepend: [{ type: core.Input }]
    };

    var NgxMasonryModule = /** @class */ (function () {
        function NgxMasonryModule() {
        }
        return NgxMasonryModule;
    }());
    NgxMasonryModule.decorators = [
        { type: core.NgModule, args: [{
                    imports: [],
                    declarations: [NgxMasonryComponent, NgxMasonryDirective],
                    exports: [NgxMasonryComponent, NgxMasonryDirective]
                },] }
    ];

    /*
     * Public API Surface of ngx-masonry
     */

    /**
     * Generated bundle index. Do not edit.
     */

    exports.NgxMasonryComponent = NgxMasonryComponent;
    exports.NgxMasonryDirective = NgxMasonryDirective;
    exports.NgxMasonryModule = NgxMasonryModule;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
