// @ngneat/content-loader/esm2020/lib/bullet-list-loader.component.mjs
import { ChangeDetectionStrategy, Component } from '@angular/core';
import { ContentLoaderComponent } from './content-loader.component';
import * as i0 from "@angular/core";
import * as i1 from "./content-loader.component";
export class BulletListContentLoaderComponent extends ContentLoaderComponent {
}
/** @nocollapse */ /** @nocollapse */ BulletListContentLoaderComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.1", ngImport: i0, type: BulletListContentLoaderComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ /** @nocollapse */ BulletListContentLoaderComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.1", type: BulletListContentLoaderComponent, selector: "bullet-list-content-loader", usesInheritance: true, ngImport: i0, template: `
    <content-loader viewBox="0 0 245 125">
      <svg:circle cx="10" cy="20" r="8" />
      <svg:rect x="25" y="15" rx="5" ry="5" width="220" height="10" />
      <svg:circle cx="10" cy="50" r="8" />
      <svg:rect x="25" y="45" rx="5" ry="5" width="220" height="10" />
      <svg:circle cx="10" cy="80" r="8" />
      <svg:rect x="25" y="75" rx="5" ry="5" width="220" height="10" />
      <svg:circle cx="10" cy="110" r="8" />
      <svg:rect x="25" y="105" rx="5" ry="5" width="220" height="10" />
    </content-loader>
  `, isInline: true, components: [{ type: i1.ContentLoaderComponent, selector: "content-loader", inputs: ["animate", "baseUrl", "speed", "viewBox", "gradientRatio", "backgroundColor", "backgroundOpacity", "foregroundColor", "foregroundOpacity", "rtl", "interval", "style"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.1", ngImport: i0, type: BulletListContentLoaderComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'bullet-list-content-loader',
                    template: `
    <content-loader viewBox="0 0 245 125">
      <svg:circle cx="10" cy="20" r="8" />
      <svg:rect x="25" y="15" rx="5" ry="5" width="220" height="10" />
      <svg:circle cx="10" cy="50" r="8" />
      <svg:rect x="25" y="45" rx="5" ry="5" width="220" height="10" />
      <svg:circle cx="10" cy="80" r="8" />
      <svg:rect x="25" y="75" rx="5" ry="5" width="220" height="10" />
      <svg:circle cx="10" cy="110" r="8" />
      <svg:rect x="25" y="105" rx="5" ry="5" width="220" height="10" />
    </content-loader>
  `,
                    changeDetection: ChangeDetectionStrategy.OnPush
                }]
        }] });
