// @ngneat/content-loader/esm2020/lib/content-loader.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { BulletListContentLoaderComponent } from './bullet-list-loader.component';
import { ContentLoaderComponent } from './content-loader.component';
import { FacebookContentLoaderComponent } from './facebook-loader.component';
import { ListContentLoaderComponent } from './list-loader.component';
import * as i0 from "@angular/core";
export class ContentLoaderModule {
    static forRoot() {
        return {
            ngModule: ContentLoaderModule
        };
    }
}
/** @nocollapse */ /** @nocollapse */ ContentLoaderModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.1", ngImport: i0, type: ContentLoaderModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ /** @nocollapse */ ContentLoaderModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.1", ngImport: i0, type: ContentLoaderModule, declarations: [ContentLoaderComponent,
        FacebookContentLoaderComponent,
        BulletListContentLoaderComponent,
        ListContentLoaderComponent], imports: [CommonModule], exports: [ContentLoaderComponent,
        FacebookContentLoaderComponent,
        BulletListContentLoaderComponent,
        ListContentLoaderComponent] });
/** @nocollapse */ /** @nocollapse */ ContentLoaderModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.1", ngImport: i0, type: ContentLoaderModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.1", ngImport: i0, type: ContentLoaderModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        ContentLoaderComponent,
                        FacebookContentLoaderComponent,
                        BulletListContentLoaderComponent,
                        ListContentLoaderComponent
                    ],
                    exports: [
                        ContentLoaderComponent,
                        FacebookContentLoaderComponent,
                        BulletListContentLoaderComponent,
                        ListContentLoaderComponent
                    ]
                }]
        }] });
