// @ngneat/content-loader/esm2020/lib/facebook-loader.component.mjs
import { ChangeDetectionStrategy, Component } from '@angular/core';
import { ContentLoaderComponent } from './content-loader.component';
import * as i0 from "@angular/core";
import * as i1 from "./content-loader.component";
export class FacebookContentLoaderComponent extends ContentLoaderComponent {
}
/** @nocollapse */ /** @nocollapse */ FacebookContentLoaderComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.1", ngImport: i0, type: FacebookContentLoaderComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ /** @nocollapse */ FacebookContentLoaderComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.1", type: FacebookContentLoaderComponent, selector: "facebook-content-loader", usesInheritance: true, ngImport: i0, template: `
    <content-loader viewBox="0 0 476 124">
      <svg:rect x="48" y="8" width="88" height="6" rx="3" />
      <svg:rect x="48" y="26" width="52" height="6" rx="3" />
      <svg:rect x="0" y="56" width="410" height="6" rx="3" />
      <svg:rect x="0" y="72" width="380" height="6" rx="3" />
      <svg:rect x="0" y="88" width="178" height="6" rx="3" />
      <svg:circle cx="20" cy="20" r="20" />
    </content-loader>
  `, isInline: true, components: [{ type: i1.ContentLoaderComponent, selector: "content-loader", inputs: ["animate", "baseUrl", "speed", "viewBox", "gradientRatio", "backgroundColor", "backgroundOpacity", "foregroundColor", "foregroundOpacity", "rtl", "interval", "style"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.1", ngImport: i0, type: FacebookContentLoaderComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'facebook-content-loader',
                    template: `
    <content-loader viewBox="0 0 476 124">
      <svg:rect x="48" y="8" width="88" height="6" rx="3" />
      <svg:rect x="48" y="26" width="52" height="6" rx="3" />
      <svg:rect x="0" y="56" width="410" height="6" rx="3" />
      <svg:rect x="0" y="72" width="380" height="6" rx="3" />
      <svg:rect x="0" y="88" width="178" height="6" rx="3" />
      <svg:circle cx="20" cy="20" r="20" />
    </content-loader>
  `,
                    changeDetection: ChangeDetectionStrategy.OnPush
                }]
        }] });
