// @ngneat/content-loader/esm2020/lib/list-loader.component.mjs
import { ChangeDetectionStrategy, Component } from '@angular/core';
import { ContentLoaderComponent } from './content-loader.component';
import * as i0 from "@angular/core";
import * as i1 from "./content-loader.component";
export class ListContentLoaderComponent extends ContentLoaderComponent {
}
/** @nocollapse */ /** @nocollapse */ ListContentLoaderComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.1", ngImport: i0, type: ListContentLoaderComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ /** @nocollapse */ ListContentLoaderComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.1", type: ListContentLoaderComponent, selector: "list-content-loader", usesInheritance: true, ngImport: i0, template: `
    <content-loader viewBox="0 0 400 110">
      <svg:rect x="0" y="0" rx="3" ry="3" width="250" height="10" />
      <svg:rect x="20" y="20" rx="3" ry="3" width="220" height="10" />
      <svg:rect x="20" y="40" rx="3" ry="3" width="170" height="10" />
      <svg:rect x="0" y="60" rx="3" ry="3" width="250" height="10" />
      <svg:rect x="20" y="80" rx="3" ry="3" width="200" height="10" />
      <svg:rect x="20" y="100" rx="3" ry="3" width="80" height="10" />
    </content-loader>
  `, isInline: true, components: [{ type: i1.ContentLoaderComponent, selector: "content-loader", inputs: ["animate", "baseUrl", "speed", "viewBox", "gradientRatio", "backgroundColor", "backgroundOpacity", "foregroundColor", "foregroundOpacity", "rtl", "interval", "style"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.1", ngImport: i0, type: ListContentLoaderComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'list-content-loader',
                    template: `
    <content-loader viewBox="0 0 400 110">
      <svg:rect x="0" y="0" rx="3" ry="3" width="250" height="10" />
      <svg:rect x="20" y="20" rx="3" ry="3" width="220" height="10" />
      <svg:rect x="20" y="40" rx="3" ry="3" width="170" height="10" />
      <svg:rect x="0" y="60" rx="3" ry="3" width="250" height="10" />
      <svg:rect x="20" y="80" rx="3" ry="3" width="200" height="10" />
      <svg:rect x="20" y="100" rx="3" ry="3" width="80" height="10" />
    </content-loader>
  `,
                    changeDetection: ChangeDetectionStrategy.OnPush
                }]
        }] });
