// ngx-star-rating/esm2020/lib/ngx-star-rating.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NgxStarRatingComponent } from './ngx-star-rating.component';
import * as i0 from "@angular/core";
export class NgxStarRatingModule {
}
NgxStarRatingModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.1", ngImport: i0, type: NgxStarRatingModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgxStarRatingModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.1", ngImport: i0, type: NgxStarRatingModule, declarations: [NgxStarRatingComponent], imports: [CommonModule], exports: [NgxStarRatingComponent] });
NgxStarRatingModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.1", ngImport: i0, type: NgxStarRatingModule, imports: [[
            CommonModule
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.1", ngImport: i0, type: NgxStarRatingModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NgxStarRatingComponent],
                    imports: [
                        CommonModule
                    ],
                    exports: [NgxStarRatingComponent]
                }]
        }] });
