// @ks89/angular-modal-gallery/esm2015/ks89-angular-modal-gallery.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
 * Generated bundle index. Do not edit.
 */
export { GalleryModule, Action, Image, ImageEvent, ImageModalEvent, DescriptionStrategy, ButtonsStrategy, ButtonType, GalleryService, LoadingType, KS_DEFAULT_ACCESSIBILITY_CONFIG, KS_DEFAULT_BTN_FULL_SCREEN, KS_DEFAULT_BTN_CLOSE, KS_DEFAULT_BTN_DELETE, KS_DEFAULT_BTN_DOWNLOAD, KS_DEFAULT_BTN_EXTURL, KS_DEFAULT_SIZE, LineLayout, GridLayout, AdvancedLayout, PlainGalleryStrategy } from './public_api';
export { AccessibleComponent as ɵi } from './lib/components/accessible.component';
export { BackgroundComponent as ɵf } from './lib/components/background/background.component';
export { CarouselPreviewsComponent as ɵn } from './lib/components/carousel/carousel-previews/carousel-previews.component';
export { CarouselComponent as ɵe } from './lib/components/carousel/carousel.component';
export { COMPONENTS as ɵc } from './lib/components/components';
export { CurrentImageComponent as ɵh } from './lib/components/current-image/current-image.component';
export { LoadingSpinnerComponent as ɵr } from './lib/components/current-image/loading-spinner/loading-spinner.component';
export { DotsComponent as ɵp } from './lib/components/dots/dots.component';
export { ModalGalleryComponent as ɵd } from './lib/components/modal-gallery/modal-gallery.component';
export { PlainGalleryComponent as ɵg } from './lib/components/plain-gallery/plain-gallery.component';
export { PreviewsComponent as ɵq } from './lib/components/previews/previews.component';
export { UpperButtonsComponent as ɵo } from './lib/components/upper-buttons/upper-buttons.component';
export { ATagBgImageDirective as ɵy } from './lib/directives/a-tag-bg-image.directive';
export { ClickOutsideDirective as ɵt } from './lib/directives/click-outside.directive';
export { DescriptionDirective as ɵz } from './lib/directives/description.directive';
export { DirectionDirective as ɵx } from './lib/directives/direction.directive';
export { DIRECTIVES as ɵs } from './lib/directives/directives';
export { KeyboardNavigationDirective as ɵv } from './lib/directives/keyboard-navigation.directive';
export { MarginDirective as ɵba } from './lib/directives/margin.directive';
export { MaxSizeDirective as ɵbb } from './lib/directives/max-size.directive';
export { SizeDirective as ɵu } from './lib/directives/size.directive';
export { WrapDirective as ɵw } from './lib/directives/wrap.directive';
export { KsHammerGestureConfig as ɵa, setupKeyboardService as ɵb } from './lib/modal-gallery.module';
export { IdValidatorService as ɵm } from './lib/services/id-validator.service';
export { KEYBOARD_CONFIGURATION as ɵj, KeyboardService as ɵk } from './lib/services/keyboard.service';
